docs/application/native/guides/device/battery-monitor.md

```diff
@@ -0,0 +1,199 @@
+    else
+        printf("Error Occurred [%d]", error_code);
+
+    ```
+
+- You can get the battery usage values for all the resources used by an application ID over a specific duration of time.
+  To get resource ID, use the `battery_monitor_resource_id_e` enum, and to get duration type information, use the `battery_monitor_duration_type_e` enum:
+
+    ```
+    battery_monitor_h data_handle = NULL;
+
+
+    /*First use this API to fetch the usage for all the available resources in the data handle*/
+    error_code = battery_monitor_get_usage_by_app_id_for_all_resource_id(app_id, duration_val, &data_handle);
+
+    if (error_code == BATTERY_MONITOR_ERROR_NONE)
+        printf("Data Handle information received");
+    else
+        printf("Error Occurred [%d]", error_code);
+
+    /*Now retrieving the information one by one from the data_handle*/
+    battery_monitor_resource_id_e resource_id = BATTERY_MONITOR_RESOURCE_ID_DISPLAY;
+
+    error_code = battery_monitor_get_usage_for_resource_id(data_handle, resource_id, &battery_usage);
+    if (error_code == BATTERY_MONITOR_ERROR_NONE)
+        printf("Battery Usage by display is [%d]", battery_usage);
+
+    resource_id = BATTERY_MONITOR_RESOURCE_ID_WIFI;
+
+    error_code = battery_monitor_get_usage_for_resource_id(data_handle, resource_id, &battery_usage)
+    if (error_code == BATTERY_MONITOR_ERROR_NONE)
+        printf("Battery Usage by display is [%d]", battery_usage);
+
+    /* Similarly for other resources...
+    .
+    .
+    .
+    */
+
+    /*To avoid meomry leak free the memory*/
+    error_code = battery_monitor_destroy(data_handle);
+    if (error_code != BATTERY_MONITOR_ERROR_NONE)
+        printf("Error Occurred [%d]", error_code);
+    ```
+
+<a name="resourceusage_get"></a>
+## Getting Battery Usage Information of Resource
+
+To get battery usage information related with resource ID:
+
+- You can get the battery usage information of a particular resource over a specific duration of time.
+  To get resource ID, use the `battery_monitor_resource_id_e` enum, and to get duration type information, use the `battery_monitor_duration_type_e` enum:
+
+    ```
+    resource_id = BATTERY_MONITOR_RESOURCE_ID_DISPLAY;
+    error_code = battery_monitor_get_total_usage_by_resource_id(resource_id, duration_val, &battery_usage);
+    if (error_code == BATTERY_MONITOR_ERROR_NONE)
+        printf("Battery Usage Display in last 24 hrs is [%d]", battery_usage);
+
+    resource_id = BATTERY_MONITOR_RESOURCE_ID_WIFI;
+    error_code = battery_monitor_get_total_usage_by_resource_id(resource_id, duration_val, &battery_usage);
+    if (error_code == BATTERY_MONITOR_ERROR_NONE)
+        printf("Battery Usage Wifi in last 24 hrs is [%d]", battery_usage);
+
+    /* Similarly for other resources...
+    .
+    .
+    .
+    */
+
+    ```
+<a name="resource_key"></a>
+## Resource Keys
+
+The following table lists the available resource keys, which are part of `battery_monitor_resource_id_e`:
```

Comment:
@d1-agarwal 
Kindly change to: 

The following table lists the available resource keys, which are part of the `battery_monitor_resource_id_e` enum: